lib/engine/route.rb

```diff
@@ -165,6 +165,22 @@ def check_connected!(token)
       # rubocop:enable Style/GuardClause, Style/IfUnlessModifier
     end
 
+    def check_terminals!
+      return unless paths.size > 2
+
+      ordered_paths = []
+      @connections.each do |c|
```

Comment:
ordered_paths = @connections.map do |c|
  paths = c[:connections].paths
  paths[0].nodes.any? { ... } // i'm not sure what your code is doing

Reply:
Yep. I think it needs to be flat_map, though.

I'm putting the paths into the order implied by the order of connections. If a connection goes between node A and node B, the paths associated with that connection can be A...B or B...A. I need to have a list of paths in exactly the order encountered going from node A to node B. This is so I can check that that none of the paths - other than the start and end paths - have the "terminal" attribute.

Reply:
Ok, pushed again. I left the variable name "cpaths" to avoid confusion with the method named paths.